src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -159,20 +159,36 @@ def launch(
 
         settings = cast(SagemakerStepOperatorSettings, self.get_settings(info))
 
-        session = sagemaker.Session(default_bucket=self.config.bucket)
-        instance_type = settings.instance_type or "ml.m5.large"
+        # Get and default fill SageMaker estimator arguments 
+        estimator_args = settings.estimator_args
+        session = estimator_args.sagemaker_session or sagemaker.Session(default_bucket=self.config.bucket)
+        instance_type = estimator_args.instance_type or "ml.m5.large"
+        instance_count = estimator_args.instance_count or 1
```

Comment:
Unfortunately this has to be fixed to `1` for the moment as the container runs some code to register entities in the ZenML database. We would first need to find a way to avoid this and/or only run it in one instance.

Reply:
Fixed to 1 in https://github.com/zenml-io/zenml/pull/1381/commits/d95b2eb2adcd97691cd49775861fceb465eb4b74.